lib/ansible/modules/monitoring/zabbix/zabbix_user.py

```diff
@@ -0,0 +1,621 @@
+            except Exception as e:
+                self._zapi.logout()
+                self._module.fail_json(msg="Failed to update user %s: %s" % (alias, e))
+
+            try:
+                user_ids = self._zapi.user.updatemedia({
+                    'users': [{'userid': zbx_user['userid']}],
+                    'medias': user_medias
+                })
+            except Exception as e:
+                self._zapi.logout()
+                self._module.fail_json(msg="Failed to update user medias %s: %s" % (alias, e))
+
+        if zbx_api_version >= 34:
+            try:
+                request_data['user_medias'] = user_medias
+                user_ids = self._zapi.user.update(request_data)
+            except Exception as e:
+                self._zapi.logout()
+                self._module.fail_json(msg="Failed to update user %s: %s" % (alias, e))
+
+        return user_ids
+
+    def delete_user(self, zbx_user, alias):
+        user_ids = {}
+        try:
+            user_ids = self._zapi.user.delete([zbx_user['userid']])
+        except Exception as e:
+            self._zapi.logout()
+            self._module.fail_json(msg="Failed to delete user %s: %s" % (alias, e))
+
+        return user_ids
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            validate_certs=dict(type='bool', required=False, default=True),
+            alias=dict(type='str', required=True),
+            name=dict(type='str', default=''),
+            surname=dict(type='str', default=''),
+            usrgrps=dict(type='list', required=True),
+            passwd=dict(type='str', required=True, no_log=True),
+            override_passwd=dict(type='bool', required=False, default=False),
+            lang=dict(type='str', default='en_GB', choices=['en_GB', 'en_US', 'zh_CN', 'cs_CZ', 'fr_FR',
+                                                            'he_IL', 'it_IT', 'ko_KR', 'ja_JP', 'nb_NO',
+                                                            'pl_PL', 'pt_BR', 'pt_PT', 'ru_RU', 'sk_SK',
+                                                            'tr_TR', 'uk_UA']),
+            theme=dict(type='str', default='default', choices=['default', 'blue-theme', 'dark-theme']),
+            autologin=dict(type='bool', default=False),
+            autologout=dict(type='str', default='0'),
+            refresh=dict(type='str', default='30'),
+            rows_per_page=dict(type='str', default='50'),
+            after_login_url=dict(type='str', default=''),
+            user_medias=dict(type='list', required=True,
```

Comment:
`user_medias` is required, but I don't see the need for it. I believe it is also not a required parameter in Zabbix API.
Is there a reason for it to be required?

Reply:
That is true for sure.
Removed required key as `user_medias` was not necessarily required.